kernel/src/process_standard.rs

```diff
@@ -448,7 +453,10 @@ impl<C: Chip> Process for ProcessStandard<'_, C> {
     }
 
     fn terminate(&self, completion_code: Option<u32>) {
-        if !self.is_running() {
+        if !self.is_running()
+            && self.get_state() != State::Faulted
+            && self.get_state() != State::CredentialsApproved
```

Comment:
Why can we not terminate a function that has had its credentials approved, but has not yet run?

Reply:
Oh, we *can* terminate it. Note the !=. The logic here is confusing, I agree. @brghena pointed this out. I have added a comment to explain.